$Header: /p/shore/shore_cvs/src/vas/server/nfs_svc.patch,v 1.2 1995/01/18 18:20:56 nhall Exp $
*** nfs_svc.c.orig	Sun Jun 21 07:23:43 1992
--- nfs_svc.c	Sun Jun 21 18:01:37 1992
***************
*** 146,152 ****
  		svcerr_decode(transp);
  		return;
  	}
! 	result = (*local)(&argument, rqstp);
  	if (result != NULL && !svc_sendreply(transp, xdr_result, result)) {
  		svcerr_systemerr(transp);
  	}
--- 146,162 ----
  		svcerr_decode(transp);
  		return;
  	}
! 	{
! 		struct authunix_parms *credentials =
! 			(struct authunix_parms *)rqstp->rq_clntcred;
! 		if (rqstp->rq_proc != NFSPROC_NULL &&
! 			(rqstp->rq_cred.oa_flavor != AUTH_UNIX || credentials == 0)) {
! 				fprintf(stderr,"bad credentials(%d,0x%x)\n",
! 					rqstp->rq_cred.oa_flavor, credentials);
! 				svcerr_weakauth(transp);
! 		}
! 		result = (*local)(&argument, credentials);
! 	}
  	if (result != NULL && !svc_sendreply(transp, xdr_result, result)) {
  		svcerr_systemerr(transp);
  	}
***************
*** 169,174 ****
--- 179,185 ----
  	char *result;
  	bool_t (*xdr_argument)(), (*xdr_result)();
  	char *(*local)();
+ 	int auth_required = 0;
  
  	switch (rqstp->rq_proc) {
  	case MOUNTPROC_NULL:
***************
*** 181,186 ****
--- 192,198 ----
  		xdr_argument = xdr_dirpath;
  		xdr_result = xdr_fhstatus;
  		local = (char *(*)()) mountproc_mnt_1;
+ 		auth_required = 1;
  		break;
  
  	case MOUNTPROC_DUMP:
***************
*** 193,198 ****
--- 205,211 ----
  		xdr_argument = xdr_dirpath;
  		xdr_result = xdr_void;
  		local = (char *(*)()) mountproc_umnt_1;
+ 		auth_required = 1;
  		break;
  
  	case MOUNTPROC_UMNTALL:
***************
*** 199,204 ****
--- 212,218 ----
  		xdr_argument = xdr_void;
  		xdr_result = xdr_void;
  		local = (char *(*)()) mountproc_umntall_1;
+ 		auth_required = 1;
  		break;
  
  	case MOUNTPROC_EXPORT:
***************
*** 222,228 ****
  		svcerr_decode(transp);
  		return;
  	}
! 	result = (*local)(&argument, rqstp);
  	if (result != NULL && !svc_sendreply(transp, xdr_result, result)) {
  		svcerr_systemerr(transp);
  	}
--- 236,252 ----
  		svcerr_decode(transp);
  		return;
  	}
! 	{
! 		struct authunix_parms *credentials =
! 			(struct authunix_parms *)rqstp->rq_clntcred;
! 		if (auth_required &&
! 			(rqstp->rq_cred.oa_flavor != AUTH_UNIX || credentials == 0)) {
! 				fprintf(stderr,"bad credentials(%d,0x%x)\n",
! 					rqstp->rq_cred.oa_flavor, credentials);
! 				svcerr_weakauth(transp);
! 		}
! 		result = (*local)(&argument, rqstp, credentials);
! 	}
  	if (result != NULL && !svc_sendreply(transp, xdr_result, result)) {
  		svcerr_systemerr(transp);
  	}
